pkg/client/clientcmd/client_config_test.go

```diff
@@ -162,7 +162,13 @@ func TestCreateCleanWithPrefix(t *testing.T) {
 		{"anything.com:8080", "anything.com:8080", ""},
 		{"anything.com", "anything.com", ""},
 		{"anything", "anything", ""},
-		{"", "http://localhost:8080", ""},
+	}
+
+	// WARNING: EnvVarCluster.Server is set during package loading time and can not be overriden by os.Setenv inside this test
+	if EnvVarCluster.Server != "" {
```

Comment:
The comment is good, but I still think this should be
```
EnvVarCluster.Server = ""
tt = append(tt, struct{ server, host, prefix string }{"", "http://localhost:8080", ""}
```

Otherwise you have a unit test that exercises different things depending on local environment. We can add a separate test for expected behavior when `EnvVarCluster.Server != ""`